### PR TITLE
Add tests for Really Object

### PR DESCRIPTION
**Pull request by ahmadajmi**
Write tests for Really Object methods `update`, `delete`, `onUpdate`, `onDelete`

### CODE REVIEW

File: tests/really-object-spec.coffee [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# Dependencies
+
+Q            = require 'q'
+ReallyObject = require '../src/really-object'
+ReallyError  = require '../src/really-error'
+protocol     = require '../src/protocol'
+Websocket    = require '../src/transports/webSocket'
+CONFIG       = require './support/server/config'
+Really       = require '../src/really'
```
Comment (NouranMahmoud) [inline tests/really-object-spec.coffee]: You no longer need Really anymore.

File: tests/really-object-spec.coffee [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+describe 'really-object', ->
+
+  describe 'get', ->
+
+    it 'should throw error if called without passing res parameter', ->
```
Comment (NouranMahmoud) [inline tests/really-object-spec.coffee]: `resource` instead of `res`

File: tests/really-object-spec.coffee [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      expect ->
+        object = new ReallyObject()
+        msg = object.get()
+      .toThrow new ReallyError('Can not be initialized without resource')
+
+    it 'should throw error if called with non-string res parameter', ->
```
Comment (NouranMahmoud) [inline tests/really-object-spec.coffee]: same here for `res`

File: tests/really-object-spec.coffee [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+      expect ->
+        object = new ReallyObject()
+        msg    = object.get(123)
+      .toThrow new ReallyError('Can not be initialized without resource')
+
+    it 'should resolve promise when message send successfully', (done) ->
```
Comment (NouranMahmoud) [inline tests/really-object-spec.coffee]: `message is sent`

File: tests/really-object-spec.coffee [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+Q            = require 'q'
+ReallyObject = require '../src/really-object'
+ReallyError  = require '../src/really-error'
+protocol     = require '../src/protocol'
+Websocket    = require '../src/transports/webSocket'
+CONFIG       = require './support/server/config'
+Really       = require '../src/really'
+
+describe 'really-object', ->
+
```
Comment (NouranMahmoud) [inline tests/really-object-spec.coffee]: You just need to mock `websocket` as we don't need to initialize real `websocket` in the tests...

try this
channel =
  send: -> true

File: tests/really-object-spec.coffee [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+        evt: 'deleted'
+        rev: 12
+        data:
+          fields: ['name': 'really']
+
+      object = new ReallyObject()
+      spyOn(object, 'on').and.callThrough()
+
+      object.onDelete(message.r, (data) ->
+        expect(object.on).toHaveBeenCalled()#With "#{message.r}:deleted", (data) -> false
```
Comment (NouranMahmoud) [inline tests/really-object-spec.coffee]: Here, you can use `jasmine.any(Function)`

File: tests/really-object-spec.coffee [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+      expect ->
+        object = new ReallyObject()
+        msg    = object.get(123)
+      .toThrow new ReallyError('Can not be initialized without resource')
+
+    it 'should resolve promise when message sent successfully', (done) ->
+      options =
+        fields: ['author']
+
+      spyOn(channel, 'send').and.callFake( () ->
```
Comment (NouranMahmoud) [inline tests/really-object-spec.coffee]: there's a very little tiny fix ... remove the `( )` which surround the `callFake(`. 
from all the file.

**Comment by NouranMahmoud** [comment]
LGTM

File: tests/really-object-spec.coffee [hunk 8]
```diff
@@ -0,0 +1,194 @@
+# Dependencies
+
+Q            = require 'q'
+ReallyObject = require '../src/really-object'
+ReallyError  = require '../src/really-error'
+protocol     = require '../src/protocol'
+
+describe 'really-object', ->
```
Comment (ikhattab) [inline tests/really-object-spec.coffee]: ReallyObject

File: tests/really-object-spec.coffee [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+    it 'should throw error if called without passing resource parameter', ->
+      expect ->
+        object = new ReallyObject()
+        msg = object.get()
+      .toThrow new ReallyError('Can not be initialized without resource')
+
+    it 'should throw error if called with non-string resource parameter', ->
+      expect ->
+        object = new ReallyObject()
+        msg    = object.get(123)
```
Comment (ikhattab) [inline tests/really-object-spec.coffee]: it'd be better if you just put what you expect to throw error in expect not the whole code; for clarity
```coffeescript
object = new ReallyObject()
expect ->
  object.get(123)
...
```

**Pull request closed without merging**